dist/common/scripts/scylla_util.py

```diff
@@ -351,6 +351,199 @@ def user_data(self):
+        "L64s": 8,
+        "L80s": 10
+    }
+
+    def __get_nvme_disks_count_from_metadata(self):
+        #storageProfile in VM metadata lacks the number of NVMEs, it's hardcoded based on VM type
+        return self.instanceToDiskCount.get(self.instance_class(), 0)
+
+    @property
+    def instancelocation(self):
+        """return the location of this instance, e.g. eastus"""
+        if self.__location is None:
+            self.__location = self.__instance_metadata("location")
+        return self.__location
+
+    @property
+    def instancezone(self):
+        """return the zone of this instance, e.g. 1"""
+        if self.__zone is None:
+            self.__zone = self.__instance_metadata("zone")
+        return self.__zone
+
+    @property
+    def instancetype(self):
+        """return the type of this instance, e.g. Standard_L8s_v2"""
+        if self.__type is None:
+            self.__type = self.__instance_metadata("vmSize")
+        return self.__type
+
+    @property
+    def cpu(self):
+        """return the # of cpus of this instance"""
+        if self.__cpu is None:
+            self.__cpu = psutil.cpu_count()
+        return self.__cpu
+
+    @property
+    def memoryGB(self):
+        """return the size of memory in GB of this instance"""
+        if self.__memoryGB is None:
+            self.__memoryGB = psutil.virtual_memory().total/1024/1024/1024
+        return self.__memoryGB
+
+    def instance_version(self):
+        """Returns the size of the instance we are running in. i.e.: v2"""
+        instancetypesplit = self.instancetype.split("_")
+        return instancetypesplit[2] if len(instancetypesplit)>2 else ""
+
+    def instance_purpose(self):
+        """Returns the class of the instance we are running in. i.e.: Standard"""
+        return self.instancetype.split("_")[0]
+
+    def instance_class(self):
+        """Returns the purpose of the instance we are running in. i.e.: L8s"""
+        return self.instancetype.split("_")[1]
+
+    def is_unsupported_instance(self):
+        """Returns if this instance type belongs to unsupported ones for nvmes"""
+        return False
+
+    def is_supported_instance(self):
+        """Returns if this instance type belongs to supported ones for nvmes"""
+        if self.instance_class() in ['L8s', 'L16s', 'L32s' ,'L48s','L64s','L80s']:
+            return True
+        return False
+
+    def is_recommended_instance_size(self):
+        """if this instance has at least 2 cpus, it has a recommended size"""
+        if int(self.instance_size()) > 1:
+            return True
+        return False
+
+    @staticmethod
+    def get_file_size_by_seek(filename):
+        "Get the file size by seeking at end"
+        fd= os.open(filename, os.O_RDONLY)
+        try:
+            return os.lseek(fd, 0, os.SEEK_END)
+        finally:
+            os.close(fd)
+
+    # as per https://docs.microsoft.com/en-us/azure/virtual-machines/lsv2-series
+    AZURE_NVME_DISK_SIZE_2020=1966
```

Comment:
Can't we query it from the disk itself?

Reply:
we do query it, this is a sanity check that we're not below this baseline with detected nvme disk